packages/headless/src/api/search/search-api-client.ts

```diff
@@ -99,14 +100,26 @@ export class SearchAPIClient {
     };
   }
 
+  private searchAbortController: AbortController | null = null;
```

Comment:
Nitpick, but I'd propose `abortSignal` for the name.

Reply:
I name it `searchAbortController` because it's the type (AbortController) and is specific to search, meaning we could have another abort controller for another method that can be problematic